core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/twitter/TwitterConverter.java

```diff
@@ -56,28 +68,33 @@ public Tuple getNextTuple() throws TexeraException {
         }
         Tuple tuple;
         while ((tuple = inputOperator.getNextTuple()) != null) {
-            List<IField> tweetFields = generateFieldsFromJson(
-                    tuple.getField(rawDataAttribute).getValue().toString());
-            if (! tweetFields.isEmpty()) {
-                cursor++;
-                
-                List<IField> tupleFields = new ArrayList<>();
-                
-                final Tuple finalTuple = tuple;
-                tupleFields.addAll(tuple.getSchema().getAttributeNames().stream()
-                        .filter(attrName -> ! attrName.equalsIgnoreCase(rawDataAttribute))
-                        .map(attrName -> finalTuple.getField(attrName, IField.class))
-                        .collect(Collectors.toList()));
-                tupleFields.addAll(tweetFields);
-                return new Tuple(outputSchema, tupleFields);
+
+            String rawTwitterJsonString = tuple.getField(this.predicate.getTwitterJsonStringFieldName()).getValue().toString();
+            Optional<List<IField>> convertedTwitterFields = this.generateFieldsFromJson(rawTwitterJsonString);
+            if (! convertedTwitterFields.isPresent()) {
+                continue;
             }
+
+            Tuple outputTuple = new Tuple.Builder(tuple)
+                // remove the raw json field
+                .remove(this.predicate.getTwitterJsonStringFieldName())
+                // add the new fields
+                .add(TwitterConverterConstants.additionalAttributes, convertedTwitterFields.get())
+                .build();
+            
+            cursor++;
+
+            return outputTuple;
         }
         return null;
     }
     
-    private List<IField> generateFieldsFromJson(String rawJsonData) {
+    private Optional<List<IField>> generateFieldsFromJson(String rawJsonData) {
```

Comment:
Instead of parsing this on our own, perhaps we can use twitter4j library in the final version of code that we write in the new backend engine (we can leave this as it is for now). That way when twitter changes its format, we just upgrade the library and everything will work. twitter4j provides API like `status.getUser().getName()` which is not affected by json format.

However, it should also be noted that twitter4j is an unofficial library (but a popular one).